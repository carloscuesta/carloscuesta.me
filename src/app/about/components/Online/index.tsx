import SectionTitle from 'src/components/shared/SectionTitle'

const profiles = [
  {
    name: 'Twitter',
    url: 'https://twitter.com/crloscuesta',
  },
  {
    name: 'GitHub',
    url: 'https://github.com/carloscuesta',
  },
  {
    name: 'LinkedIn',
    url: 'https://www.linkedin.com/in/crloscuesta/',
  },
  {
    name: 'Email',
    url: 'mailto:hi@carloscuesta.me',
  },
]

const Online = () => (
  <section>
    <SectionTitle title="Online" />

    <ul className="grid grid-cols-2 gap-3 sm:grid-flow-col sm:grid-cols-[repeat(4,min-content)]">
      {profiles.map((profile) => (
        <li key={profile.name}>
          <a
            href={profile.url}
            className="hover:text-neutral-1000 shadow-xs flex flex-none cursor-pointer items-center justify-center rounded-md border border-neutral-400 border-opacity-30 bg-transparent px-4 py-2 text-sm font-semibold leading-none text-neutral-700 opacity-100 transition-all hover:border-opacity-50 hover:shadow-sm dark:border-neutral-700 dark:text-neutral-200 dark:hover:border-neutral-600 dark:hover:text-white sm:space-x-2"
            target="_blank"
            rel="noopener noreferrer"
          >
            {profile.name}
          </a>
        </li>
      ))}
    </ul>
  </section>
)

export default Online
