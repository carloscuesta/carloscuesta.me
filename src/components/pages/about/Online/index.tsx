import SectionTitle from 'src/components/shared/SectionTitle'

const profiles = [
  {
    name: 'Twitter',
    url: 'https://twitter.com/crloscuesta',
  },
  {
    name: 'GitHub',
    url: 'https://github.com/carloscuesta',
  },
  {
    name: 'LinkedIn',
    url: 'https://www.linkedin.com/in/crloscuesta/'
  },
  {
    name: 'Email',
    url: 'mailto:hi@carloscuesta.me'
  }
]

const Online = () => (
  <section>
    <SectionTitle title='Online' />

    <ul className='grid grid-cols-2 gap-3 sm:grid-flow-col sm:grid-cols-[repeat(4,min-content)]'>
      {profiles.map((profile) => (
        <li key={profile.name}>
          <a href={profile.url} className="flex sm:space-x-2 flex-none items-center justify-center cursor-pointer leading-none transition-all font-semibold px-4 py-2 text-sm opacity-100 rounded-md text-gray-700 hover:text-gray-1000 shadow-xs bg-transparent border border-gray-400 border-opacity-30 dark:border-gray-700 dark:hover:border-gray-600 dark:text-gray-200 dark:hover:text-white hover:border-opacity-50 hover:shadow-sm" target='_blank' rel='noopener noreferrer'>{profile.name}</a>
        </li>
      ))}
    </ul>
  </section>
)

export default Online
