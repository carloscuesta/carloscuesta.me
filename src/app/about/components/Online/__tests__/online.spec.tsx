import renderer from 'react-test-renderer'

import Online from '../'

describe('Online', () => {
  it('should render the component', () => {
    const wrapper = renderer.create(<Online />)

    expect(wrapper).toMatchSnapshot()
  })
})
