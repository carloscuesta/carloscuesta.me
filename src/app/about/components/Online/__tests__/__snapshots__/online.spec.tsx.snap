// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Online should render the component 1`] = `
<section>
  <div
    className="my-6 flex items-end justify-between"
  >
    <div
      className="space-y-2"
    >
      <h2
        className="text-lg font-bold"
      >
        Online
      </h2>
    </div>
  </div>
  <ul
    className="grid grid-cols-2 gap-3 sm:grid-flow-col sm:grid-cols-[repeat(4,min-content)]"
  >
    <li>
      <a
        className="hover:text-neutral-1000 shadow-xs flex flex-none cursor-pointer items-center justify-center rounded-md border border-neutral-400 border-opacity-30 bg-transparent px-4 py-2 text-sm font-semibold leading-none text-neutral-700 opacity-100 transition-all hover:border-opacity-50 hover:shadow-sm dark:border-neutral-700 dark:text-neutral-200 dark:hover:border-neutral-600 dark:hover:text-white sm:space-x-2"
        href="https://twitter.com/crloscuesta"
        rel="noopener noreferrer"
        target="_blank"
      >
        Twitter
      </a>
    </li>
    <li>
      <a
        className="hover:text-neutral-1000 shadow-xs flex flex-none cursor-pointer items-center justify-center rounded-md border border-neutral-400 border-opacity-30 bg-transparent px-4 py-2 text-sm font-semibold leading-none text-neutral-700 opacity-100 transition-all hover:border-opacity-50 hover:shadow-sm dark:border-neutral-700 dark:text-neutral-200 dark:hover:border-neutral-600 dark:hover:text-white sm:space-x-2"
        href="https://github.com/carloscuesta"
        rel="noopener noreferrer"
        target="_blank"
      >
        GitHub
      </a>
    </li>
    <li>
      <a
        className="hover:text-neutral-1000 shadow-xs flex flex-none cursor-pointer items-center justify-center rounded-md border border-neutral-400 border-opacity-30 bg-transparent px-4 py-2 text-sm font-semibold leading-none text-neutral-700 opacity-100 transition-all hover:border-opacity-50 hover:shadow-sm dark:border-neutral-700 dark:text-neutral-200 dark:hover:border-neutral-600 dark:hover:text-white sm:space-x-2"
        href="https://www.linkedin.com/in/crloscuesta/"
        rel="noopener noreferrer"
        target="_blank"
      >
        LinkedIn
      </a>
    </li>
    <li>
      <a
        className="hover:text-neutral-1000 shadow-xs flex flex-none cursor-pointer items-center justify-center rounded-md border border-neutral-400 border-opacity-30 bg-transparent px-4 py-2 text-sm font-semibold leading-none text-neutral-700 opacity-100 transition-all hover:border-opacity-50 hover:shadow-sm dark:border-neutral-700 dark:text-neutral-200 dark:hover:border-neutral-600 dark:hover:text-white sm:space-x-2"
        href="mailto:hi@carloscuesta.me"
        rel="noopener noreferrer"
        target="_blank"
      >
        Email
      </a>
    </li>
  </ul>
</section>
`;
